echopype/echodata/simrad.py

```diff
@@ -0,0 +1,233 @@
+            # variable in Beam_group1
+            if waveform_mode == "BB" and "frequency_start" not in echodata["Sonar/Beam_group1"]:
+                raise RuntimeError("waveform_mode='BB', but broadband data not found!")
+            elif "backscatter_i" not in echodata["Sonar/Beam_group1"].variables:
+                raise RuntimeError("waveform_mode='BB', but complex data does not exist!")
+            elif echodata["Sonar/Beam_group2"] is not None:
+                power_ed_group = "Sonar/Beam_group2"
+                complex_ed_group = "Sonar/Beam_group1"
+            else:
+                complex_ed_group = "Sonar/Beam_group1"
+
+        else:
+
+            # CW can have complex or power data, so we just need to make sure that
+            # 1) complex samples always exist in Sonar/Beam_group1
+            # 2) power samples are in Sonar/Beam_group1 if only one beam group exists
+            # 3) power samples are in Sonar/Beam_group2 if two beam groups exist
+            if echodata["Sonar/Beam_group2"] is None:
+
+                if encode_mode == "power":
+                    # power samples must be in Sonar/Beam_group1 (thus no complex samples)
+                    if "backscatter_i" in echodata["Sonar/Beam_group1"].variables:
+                        raise RuntimeError(
+                            "Data provided does not correspond to encode_mode='power'!"
+                        )
+                    else:
+                        power_ed_group = "Sonar/Beam_group1"
+                elif encode_mode == "complex":
+                    # complex samples must be in Sonar/Beam_group1
+                    if "backscatter_i" not in echodata["Sonar/Beam_group1"].variables:
+                        raise RuntimeError(
+                            "Data provided does not correspond to encode_mode='complex'!"
+                        )
+                    else:
+                        complex_ed_group = "Sonar/Beam_group1"
+
+            else:
+
+                # complex should be in Sonar/Beam_group1 and power in Sonar/Beam_group2
+                # the RuntimeErrors below should never be triggered
+                if "backscatter_i" not in echodata["Sonar/Beam_group1"].variables:
+                    raise RuntimeError(
+                        "Complex data does not exist in Sonar/Beam_group1, "
+                        "input echodata object must be incorrectly constructed!"
+                    )
+                elif "backscatter_r" not in echodata["Sonar/Beam_group2"].variables:
+                    raise RuntimeError(
+                        "Power data does not exist in Sonar/Beam_group2, "
+                        "input echodata object must be incorrectly constructed!"
+                    )
+                else:
+                    complex_ed_group = "Sonar/Beam_group1"
+                    power_ed_group = "Sonar/Beam_group2"
+
+    return power_ed_group, complex_ed_group
+
+
+def check_waveform_encode_mode(
```

Comment:
since this function returns the current group path/str to be used, how about `retrieve_correct_beam_group`? I know this does not include the checking part, but it seems natural to check for input argument before loading something, so omitting "check" in the function name seems fine?